### ISSUE TITLE
Can't decompress data which is compressed by rust code.

### ISSUE THREAD

**Issue by WindSoilder** [sergey-dryabzhinsky/python-zstd#100]
Here is simplest reproducing example:
```python
import zstd
zstd.decompress(b'\x28\xb5\x2f\xfd\x00\x58\x11\x00\x00\x7b\x7d')
```
It raises an error: `Error: Input data invalid or missing content size in frame header.`

## More context
I'm trying to rewrite a client application in rust, it sends compressed data to server, then server decompresses it.  Unfortunally the server failed to decompress data.

Here is how I do it in client side:
```rust
use std::io::Cursor;
use zstd;

fn main() {
    let body = zstd::encode_all(Cursor::new("{}".as_bytes()), 3).unwrap();
    for x in body.iter() {
        print!("\\x{x:x?}");
    }
}
```
And I copied the body and decompressed it in python, and it failed.

--------

If I tried to compress data(`{}` in my example) in python, and decompressed in rust, it successes.  So I think it's the issue in python side.

In rust, I'm using [zstd-rs](https://github.com/gyscos/zstd-rs) for compressing/decompressing

**Comment by sergey-dryabzhinsky** [comment]
try to not skip first 4 bytes

пн, 12 авг. 2024 г. в 10:50, Wind ***@***.***>:

**Comment by gyscos** [comment]
It looks like python-zstd is unable to decompress zstd data when the content-size is missing from the frame header.

https://github.com/sergey-dryabzhinsky/python-zstd/blob/master/src/python-zstd.c#L157C1-L157C96

The C library (or the rust binding) do not require this, or include it by default, which is why data compressed by the rust binding cannot currently be decompressed by python-zstd.

**Comment by sergey-dryabzhinsky** [comment]
What original data is?

**Comment by sergey-dryabzhinsky** [comment]
Can you decompress data using cli version of zstd?

**Comment by WindSoilder** [comment]
Hi, the original data is `{}`.  And I can decompress data using cli version of zstd.

**Comment by sergey-dryabzhinsky** [comment]
It's clearly no valid zstd block, there is no header, but maybe it's zstd stream chunk.

**Comment by sergey-dryabzhinsky** [comment]
Yeah, it was zstd stream data. Will be possible to decompress in 1.5.6.8 version.

**Comment by sergey-dryabzhinsky** [comment]
Try new release 1.5.6.8
